src/common/models/grant.model.ts

```diff
@@ -0,0 +1,17 @@
+import { Action } from 'common/enums';
+import { BaseModel } from './base.model';
+
+export class GrantModel extends BaseModel<GrantModel> {
+  subject!: string;
+  action!: Action;
+  object!: string;
+  field?: string[];
+  filter?: string[];
+  location?: string[];
+  time?: GrantTimeModel[];
+}
+
+export class GrantTimeModel {
```

Comment:
Due to the weakness of `GrantTimeModel`, do you need it outside the SDK?
if not remove the `export` keyword.